tools/rebuild_provision_request.rb

```diff
@@ -0,0 +1,305 @@
+  end
+
+  def general_tab(_tab, dialog_tab)
+    request = @output["vm_fields"]
+    dialog_field_values(dialog_tab) do |field, value|
+      request[field] = value
+    end
+  end
+
+  def customize(tab, dialog_tab)
+    @provision_options.delete(:customization_template_script)
+    general_tab(tab, dialog_tab)
+  end
+
+  def schedule(tab, dialog_tab)
+    if @provision_options[:schedule_type].first == "immediately"
+      @provision_options.delete(:schedule_type)
+      @provision_options.delete(:schedule_time)
+    else
+      @provision_options[:schedule_time] = @provision_options[:schedule_time].to_s
+    end
+    general_tab(tab, dialog_tab)
+  end
+end
+
+class Dialog
+  attr_reader :provision_options
+
+  def initialize(provision_options, verbose)
+    @verbose = verbose
+    @provision_options = provision_options
+    load_dialog_content
+  end
+
+  def load_dialog_content
+    dialog_name = @provision_options.delete(:miq_request_dialog_name)
+    @dialog = MiqDialog.find_by(:name => dialog_name)
+  end
+
+  def content
+    @dialog.content
+  end
+end
+
+class AutomateHash
+  attr_reader :request_id
+
+  URL_PATH = "/api/provision_requests".freeze
+
+  def initialize(request_id, opts)
+    @verbose = opts[:verbose]
+    @options = opts
+    @request_id = request_id
+  end
+
+  def last_requests
+    print "\nSearching for the last #{@options[:last_requests]} requests run against #{build_url}\n\n" if @verbose
+    list = MiqProvisionRequest.limit(@options[:last_requests]).order(:created_on => 'DESC').collect(&:id)
+    if @verbose
+      print "Found these #{list.count} request ids: #{list.join(',')}\n\n"
+    else
+      puts list.join(',')
+    end
+  end
+
+  def perform
+    if @verbose
+      print "\nRunning with the following options\n\n"
+      pp @options
+    end
+    hash_output
+    console if @options[:console]
+    run_it if @options[:run_it]
+  end
+
+  def run_it
+    puts "RE Running the Rest API POST to request_id: #{@options[:request_id]}" if @verbose
+    result = RestClient.post build_url, output.to_json
+    puts result
+  end
+
+  def console
+    print "\nPlace in a Rails console to run the POST API call\n" if @verbose
+    print "\nRestClient.post '#{build_url}', '#{output.to_json}'\n"
+  end
+
+  def hash_output
+    print log if @verbose
+    puts ""
+    if @options[:output] == 'json'
+      puts JSON.pretty_generate(output)
+    else
+      pp output
+    end
+  end
+
+  def provision_options
+    find_request_options
+    @provision_options.options.dup
+  end
+
+  def dialog
+    Dialog.new(provision_options, @verbose)
+  end
+
+  def tabs
+    Tab.new(dialog.content, provision_options, @verbose)
+  end
+
+  def output
+    @output ||= tabs.output
+  end
+
+  def log
+    @log ||= tabs.log
+  end
+
+  private
+
+  def build_url
+    protocol = @options[:ssl] ? 'https' : 'http'
+    @url ||= "#{protocol}://#{@options[:username]}:#{@options[:password]}@#{@options[:api_host]}:#{@options[:port]}#{URL_PATH}"
```

Comment:
@syncrou This should probably use `URI::HTTP.build` / `URI::HTTPS.build`.  You could setup the hash first and pass it to the proper class depending on the `:ssl` setting.

Something like:
```ruby
@url ||= begin
  url_hash = {:userinfo => "#{@options[:username]}:#{@options[:password]}", :host => ...}
  @options[:ssl] ? URI::HTTPS.build(url_hash).to_s : URI::HTTP.build(url_hash).to_s
end
```
http://ruby-doc.org/stdlib-2.1.2/libdoc/uri/rdoc/URI/HTTP.html#method-c-build

There might be a better way to do this, but we should not have to manually construct the URL.